server/accounts.go

```diff
@@ -4114,18 +4120,64 @@ type transform struct {
 	src, dest string
 	dtoks     []string
 	stoks     []string
-	dtpi      []int8
+	dtpi      [][]int // destination token position indexes
+	dtpinp    []int32 // destination token position index number of partitions
+}
+
+func getMappingFunctionArgs(functionRegEx *regexp.Regexp, token string) ([]string, error) {
+	commandStrings := functionRegEx.FindStringSubmatch(token)
+	if len(commandStrings) > 1 {
+		args := commaSeparatorRegEx.Split(commandStrings[1], -1)
+		return args, nil
+	}
+	return nil, nil
 }
 
-// Helper to pull raw place holder index. Returns -1 if not a place holder.
-func placeHolderIndex(token string) int {
-	if len(token) > 1 && token[0] == '$' {
-		var tp int
-		if n, err := fmt.Sscanf(token, "$%d", &tp); err == nil && n == 1 {
-			return tp
+// Helper to pull raw place holder indexes and number of partitions. Returns -1 if not a place holder.
+func placeHolderIndex(token string) ([]int, int32) {
```

Comment:
Right now it errors out using an ErrBadSubject which is not really right because the problem is not really with the subject (that you match on) but with interpreting the destination mapping (not really a subject per se). So I think I will add a new Error that will better help.

Reply:
Will also improve the error message that ends up being logged.

For example, used to be `Error adding mapping for "foo.*.*": invalid subject` and is now `Error adding mapping for "foo.*.*" to "foo.$1.{{wildcard(2)}}.{{partition(10,1,2,a)}}" : invalid subject mapping destination`